src/init/chromeStorageHandlers.ts

```diff
@@ -0,0 +1,22 @@
+import * as chromeKeys from '../chrome-storage/keys';
+import { changeAppModes } from '../store/appModes/actions';
+import store from '../store';
+import { AppModeReducer } from '../store/appModes/reducers';
+import chromeStorage from 'chrome-storage';
+
+const storageKeysToAppMode = {
+  [chromeKeys.DISABLED_EXTENSION]: 'isExtensionDisabled',
+  [chromeKeys.ANNOTATION_MODE_PAGES]: 'annotationModePages',
+  [chromeKeys.DISABLED_PAGES]: 'disabledPages',
+};
+
+export default function initializeChromeStorageHandlers() {
+  chromeStorage.onChanged.addListener((changes, namespace) => {
+    const newModes = {};
+    for (const key of Object.keys(changes)) {
+      store.dispatch(
+        changeAppModes({ [storageKeysToAppMode[key]]: changes[key].newValue } as AppModeReducer),
```

Comment:
maybe it'd be better to construct `newModes` as
```
newModes = Object.keys(changes).reduce(acc, key => acc[key] = changes[key].newValue, {})
```
and then use a single `store.dispatch` call with the whole object?

Now there is a separate dispatch for every `key` available in `changes`